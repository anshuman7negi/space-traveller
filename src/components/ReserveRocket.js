import React from 'react';
import { useSelector } from 'react-redux';

const ReserveRocket = () => {
  const rocketData = useSelector((state) => state.rockets.rockets);

  return (
    <div className="rocketDetails">
      <h3>My Rockets</h3>
      <table>
        <tbody>
          {rocketData.map((item) => (
            item.reserved && (
              <tr key={item.id}>
                <td>{item.name}</td>
              </tr>
            )
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default ReserveRocket;
