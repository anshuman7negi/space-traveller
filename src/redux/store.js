import { configureStore } from '@reduxjs/toolkit';
import missionReducer from './missions/missionSlice';

const store = configureStore({
  reducer: {
    missions: missionReducer,
  },
});

export default store;
