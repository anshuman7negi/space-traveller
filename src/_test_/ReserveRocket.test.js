import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import ReserveRocket from '../components/ReserveRocket';

test('ReserveRocket component matches snapshot', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <ReserveRocket />
    </Provider>,
  );

  expect(asFragment()).toMatchSnapshot();
});
