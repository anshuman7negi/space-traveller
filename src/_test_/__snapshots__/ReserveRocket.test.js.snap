// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile Component MyProfile component renders correctly 1`] = `
<DocumentFragment>
  <section
    class="px-[5rem] h-auto py-[2rem] font-manrope flex justify-between items-start gap-x-[3rem]"
  >
    <div
      class="w-1/2 h-auto"
    >
      <h2
        class="text-3xl font-bold mb-3"
      >
        My missions
      </h2>
      <ul
        class="border rounded-md list-none"
      >
        <li
          class="pb-8 pt-4 pl-8 border-b font-normal"
        >
          Mission 1
        </li>
      </ul>
      <div />
    </div>
    <div
      class="w-1/2 h-auto"
    >
      <h2
        class="text-3xl font-bold mb-3"
      >
        My Rockets
      </h2>
      <ul
        class="border rounded-md list-none"
      >
        <li
          class="pb-8 pt-4 pl-8 border-b font-normal"
        >
          Rocket 1
        </li>
      </ul>
    </div>
  </section>
</DocumentFragment>
`;
